tools/cgeist/driver.cc

```diff
@@ -763,20 +811,40 @@ int main(int argc, char **argv) {
         options.dataLayout = DL;
         // invalid for gemm.c init array
         // options.useBarePtrCallConv = true;
-        pm3.addPass(
-            polygeist::createConvertPolygeistToLLVMPass(options, CStyleMemRef));
-        // pm3.addPass(mlir::createLowerFuncToLLVMPass(options));
+
+#if POLYGEIST_ENABLE_CUDA
+        if (EmitCuda) {
+          pm3.addPass(polygeist::createConvertPolygeistToLLVMPass(
+              options, CStyleMemRef, /* onlyGpuModules */ true));
+          mlir::OpPassManager &gpuPM = pm3.nest<gpu::GPUModuleOp>();
+
+          std::string arch = CUDAGPUArch;
+          if (arch == "")
+            arch = "sm_60";
+          // TODO use the clang cuda toolchain or CudaInstallationDetector to
+          // get the ptxas and libdevice paths
+          gpuPM.addPass(polygeist::createGpuSerializeToCubinPass(
+              gpuTriple.getTriple(), arch, "+ptx74", optLevel,
+              "/usr/local/cuda/bin/ptxas",
```

Comment:
Can we get this by asking clang? It should already know the path to cuda from command line args or a default.